server/src/test/java/io/crate/lucene/NumericEqQueryTest.java

```diff
@@ -115,6 +118,39 @@ public void test_numeric_comparisions_with_different_precision() {
         assertThat(convert(col + " <= -1")).isEqualTo(convert(col + " < -0.99"));
     }
 
+    @Test
+    public void test_equals_with_different_precision() {
+        String col = randomBoolean() ? "x" : "y";
+        assertThat(convert(col + " = 1")).isEqualTo(convert(col + " = 1.00"));
+        assertThat(convert(col + " = 1.1")).isEqualTo(convert(col + " = 1.10"));
+        // TODO the cast is required to preserve the precision/scale until NumericEqQuery
+        assertThat(convert(col + " = 1.111::numeric")).isExactlyInstanceOf(MatchNoDocsQuery.class);
+        assertThat(convert(col + " = 1.110::numeric")).isEqualTo(convert(col + " = 1.11"));
+    }
+
+    @Test
+    public void test_incomparable_numeric_values_are_filtered_from_numeric_array_literal() {
+        // TODO the cast is required to preserve the precision/scale until NumericEqQuery
+        assertThat(convert("xarr = [1, 1.1, 1.110, 1.111]::numeric[]").toString())
+            .isEqualTo("+xarr:{100 110 111} +(xarr = [1.0, 1.1, 1.11, 1.111])"); // generic query still includes 1.111
+        assertThat(convert("yarr = [1, 1.1, 1.110, 1.111]::numeric[]").toString())
+            .isEqualTo("+yarr:{100 110 111} +(yarr = [1.0, 1.1, 1.11, 1.111])");
+        assertThat(convert("xarr = [1.111]::numeric[]").toString())
+            .isEqualTo("+MatchNoDocsQuery(\"The given values are out of bounds\") +(xarr = [1.111])");
+        assertThat(convert("yarr = [1.111]::numeric[]").toString())
+            .isEqualTo("+MatchNoDocsQuery(\"The given values are out of bounds\") +(yarr = [1.111])");
+    }
+
+    @Test
+    public void test_same_significant_digits() {
+        String col = randomBoolean() ? "x" : "y";
+        assertThat(convert(col + " = 1.11::numeric").toString()).isEqualTo(col + ":[111 TO 111]");
+        assertThat(convert(col + " = 11.1::numeric").toString()).isEqualTo(col + ":[1110 TO 1110]");
+        assertThat(convert(col + " = 111::numeric").toString()).isEqualTo(col + ":[11100 TO 11100]");
+        assertThat(convert(col + "arr = [1.11, 11.1, 111]::numeric[]").toString())
```

Comment:
Looks like this relates to a different fix than the assertions above, if so, please move to a dedicated test case (and commit).